src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerFileReference.cs

```diff
@@ -462,16 +462,15 @@ public bool Equals(AnalyzerReference other)
         {
             if (other != null)
             {
-                return other.Display == this.Display &&
-                       other.FullPath == this.FullPath;
+                return other.FullPath == this.FullPath;
```

Comment:
>other.FullPath == this.FullPath; [](start = 23, length = 32)

Should we include _assemblyLoader into the comparison?

Reply:
_Possibly_, unfortunately doing so might mean the check becomes worthless if people are creating an IAnalyzerAssemblyLoader per reference. I suspect minimal change here is probably safest.

Reply:
Sorry, it looks like both ```IAnalyzerAssemblyLoader``` and ```AnalyzerFileReference``` are public APIs. We have no Idea how ```IAnalyzerAssemblyLoader``` is implemented and whether two distinct instances of ```IAnalyzerAssemblyLoader``` are going to have an equivalent result. The only safe way to check that is to compare the two. Even if distinct instances are passed to different ```AnalyzerFileReference``` objects, they should simply say that they are equal if they do the same thing.

Reply:
Have we previously ever required IAnalyzerAssemblyLoader (which is an extensibility point potentially implemented by third parties) to define equality? Adding a check is most definitely _not_ safe if we haven't, since we've just created a new contract and a breaking change.

Reply:
> Adding a check is most definitely not safe if we haven't, since we've just created a new contract and a breaking change.

This is true for any change we make here, including the one you already made. The implemented "value" equality doesn't really make sense to me because it doesn't guarantee equivalence in behavior of two instances. What is the point of doing this if one can always get the path and compare it the way which is appropriate (case-sensitive vs. case insensitive, etc.) If it doesn't make sense to have a concept of "value" equality for this class, then we shouldn't claim to implement one.

Reply:
I agree in hindsight it doesn't make sense for this to try to implement equality at all. But the reason I'm here in the first place is code is code does exist that is using equality on this, and this change improves the performance there without making additional assumptions about how that code works.

My change has a demonstrable benefit to the end user and a risk profile that is well understood. Adding the assembly loader has the benefit of meeting some platonic ideal of equality, but I don't see a concrete benefit. And it creates more risk.